pennylane/math/utils.py

```diff
@@ -269,7 +269,7 @@ def _get_interface_of_single_tensor(tensor):
     """
     namespace = tensor.__class__.__module__.split(".")[0]
 
-    if namespace in ("pennylane", "autograd"):
+    if namespace == "autograd":
         return "autograd"
 
     res = ar.infer_backend(tensor)

```

Comment:
Changed this because `ar.infer_backend` dispatches `"pennylane"` to `"autograd"` (or to `"jax"` if program capture is enabled with the changes in this PR), and `infer_backend` additionally uses `functools.lru_cache`, which would give us (maybe negligible?) speedups.